### ISSUE TITLE
Double Task generated on OSx

### ISSUE THREAD

**Issue by Burden50** [gboudreau/Greyhole#196]
It seems when a file is copied from OSX to my greyhole share it generates 2 tasks. 
```
Aug 08 04:18:04 DEBUG sleep: Nothing to do... Sleeping.
Aug 08 04:18:14 DEBUG sleep: Nothing to do... Sleeping.
Aug 08 04:18:24 DEBUG read_smb_spool: Processing Samba spool...
Aug 08 04:18:25 DEBUG read_smb_spool: Found 3 new tasks in spool.

Aug 08 04:18:25 INFO write: Now working on task ID 111: write TV/abcdef.txt
Aug 08 04:18:25 INFO write: File created: TV/abcdef.txt - 4.00B
Aug 08 04:18:25 DEBUG write: Loading metafiles for TV/abcdef.txt ...
Aug 08 04:18:25 DEBUG write:   Got 0 metadata files.
Aug 08 04:18:25 DEBUG write: 0 metafiles loaded.
Aug 08 04:18:25 DEBUG write:   Drives with available space: /mnt/a1/d4 (4.46TB avail) - /mnt/a1/d2 (4.46TB avail) - /mnt/a1/d3 (4.46TB avail) - /mnt/a2/d2 (4.46TB avail) - /mnt/a1/d1 (4.46TB avail) - /mnt/a2/d1 (4.46TB avail) - /mnt/a2/d4 (4.46TB avail) - /mnt/a2/d3 (4.46TB avail) - /mnt/a3/d2 (1.69TB avail) - /mnt/a3/d3 (1.69TB avail) - /mnt/a3/d4 (1.69TB avail) - /mnt/a3/d1 (1.69TB avail) 
Aug 08 04:18:25 DEBUG write:   Saving 2 metadata files for TV/abcdef.txt
Aug 08 04:18:25 DEBUG write:     Saving metadata in /mnt/a1/d2/.gh_metastore/TV/abcdef.txt
Aug 08 04:18:25 DEBUG write:     Saving metadata in /mnt/a1/d4/.gh_metastore/TV/abcdef.txt
Aug 08 04:18:25 DEBUG write:   Copying 4.00B file to /mnt/a1/d2/TV/abcdef.txt
Aug 08 04:18:25 DEBUG write:     Creating symlink in share pointing to the above file copy.
Aug 08 04:18:25 DEBUG write:   Saving 2 metadata files for TV/abcdef.txt
Aug 08 04:18:25 DEBUG write:     Saving metadata in /mnt/a1/d2/.gh_metastore/TV/abcdef.txt
Aug 08 04:18:25 DEBUG write:     Saving metadata in /mnt/a1/d4/.gh_metastore/TV/abcdef.txt
Aug 08 04:18:25 DEBUG write:   Copying 4.00B file to /mnt/a1/d4/TV/abcdef.txt
Aug 08 04:18:25 DEBUG write:   Saving 2 metadata files for TV/abcdef.txt
Aug 08 04:18:25 DEBUG write:     Saving metadata in /mnt/a1/d2/.gh_metastore/TV/abcdef.txt
Aug 08 04:18:25 DEBUG write:     Saving metadata in /mnt/a1/d4/.gh_metastore/TV/abcdef.txt

Aug 08 04:18:25 INFO unlink: Now working on task ID 112: unlink TV/abcdef.txt
Aug 08 04:18:25 INFO unlink: File deleted: /mnt/lz/TV/abcdef.txt
Aug 08 04:18:25 DEBUG unlink:   File still exists in landing zone; a new file replaced the one deleted here. Skipping.

Aug 08 04:18:25 INFO write: Now working on task ID 113: write TV/abcdef.txt
Aug 08 04:18:25 INFO write: File changed: TV/abcdef.txt - 4.00B
Aug 08 04:18:25 DEBUG write:   Will use source file: /mnt/a1/d2/TV/abcdef.txt
Aug 08 04:18:25 DEBUG write: Loading metafiles for TV/abcdef.txt ...
Aug 08 04:18:25 DEBUG write:   Got 2 metadata files.
Aug 08 04:18:25 DEBUG write:   Will remove copy at /mnt/a1/d4/TV/abcdef.txt
Aug 08 04:18:25 DEBUG write:   Moved copy from /mnt/a1/d4/TV/abcdef.txt to trash: /mnt/a1/d4/.gh_trash/TV/abcdef.txt
Aug 08 04:18:25 DEBUG write:   Drives with available space: /mnt/a1/d4 (4.46TB avail) - /mnt/a1/d2 (4.46TB avail) - /mnt/a1/d3 (4.46TB avail) - /mnt/a2/d2 (4.46TB avail) - /mnt/a1/d1 (4.46TB avail) - /mnt/a2/d1 (4.46TB avail) - /mnt/a2/d4 (4.46TB avail) - /mnt/a2/d3 (4.46TB avail) - /mnt/a3/d2 (1.69TB avail) - /mnt/a3/d3 (1.69TB avail) - /mnt/a3/d4 (1.69TB avail) - /mnt/a3/d1 (1.69TB avail) 
Aug 08 04:18:25 DEBUG write:   Saving 2 metadata files for TV/abcdef.txt
Aug 08 04:18:25 DEBUG write:     Saving metadata in /mnt/a1/d2/.gh_metastore/TV/abcdef.txt
Aug 08 04:18:25 DEBUG write:     Saving metadata in /mnt/a1/d4/.gh_metastore/TV/abcdef.txt
Aug 08 04:18:25 DEBUG write:   Copying 4.00B file to /mnt/a1/d2/TV/abcdef.txt
Aug 08 04:18:25 DEBUG write:   (using rename)
Aug 08 04:18:25 DEBUG write:     Creating symlink in share pointing to the above file copy.
Aug 08 04:18:25 DEBUG write:   Saving 2 metadata files for TV/abcdef.txt
Aug 08 04:18:25 DEBUG write:     Saving metadata in /mnt/a1/d2/.gh_metastore/TV/abcdef.txt
Aug 08 04:18:25 DEBUG write:     Saving metadata in /mnt/a1/d4/.gh_metastore/TV/abcdef.txt
Aug 08 04:18:25 DEBUG write:   Copying 4.00B file to /mnt/a1/d4/TV/abcdef.txt
Aug 08 04:18:25 DEBUG write:   Saving 2 metadata files for TV/abcdef.txt
Aug 08 04:18:25 DEBUG write:     Saving metadata in /mnt/a1/d2/.gh_metastore/TV/abcdef.txt
Aug 08 04:18:25 DEBUG write:     Saving metadata in /mnt/a1/d4/.gh_metastore/TV/abcdef.txt
Aug 08 04:18:25 DEBUG read_smb_spool: Processing Samba spool...
Aug 08 04:18:25 DEBUG sleep: Nothing to do... Sleeping.
Aug 08 04:18:35 DEBUG read_smb_spool: Processing Samba spool...
Aug 08 04:18:35 DEBUG sleep: Nothing to do... Sleeping.
```

When I connect to the smb share from linux and copy a file it works fine only generating 1 task 
```

Aug 08 04:22:47 DEBUG sleep: Nothing to do... Sleeping.
Aug 08 04:22:57 DEBUG sleep: Nothing to do... Sleeping.
Aug 08 04:23:07 DEBUG sleep: Nothing to do... Sleeping.
Aug 08 04:23:17 DEBUG read_smb_spool: Processing Samba spool...
Aug 08 04:23:17 DEBUG read_smb_spool: Found 1 new tasks in spool.
Aug 08 04:23:17 INFO write: Now working on task ID 114: write TV/1234
Aug 08 04:23:17 INFO write: File created: TV/1234 - 0.00B
Aug 08 04:23:17 DEBUG write: Loading metafiles for TV/1234 ...
Aug 08 04:23:17 DEBUG write:   Got 0 metadata files.
Aug 08 04:23:17 DEBUG write: 0 metafiles loaded.
Aug 08 04:23:17 DEBUG write:   Drives with available space: /mnt/a1/d1 (4.46TB avail) - /mnt/a1/d3 (4.46TB avail) - /mnt/a2/d2 (4.46TB avail) - /mnt/a2/d1 (4.46TB avail) - /mnt/a2/d4 (4.46TB avail) - /mnt/a1/d2 (4.46TB avail) - /mnt/a1/d4 (4.46TB avail) - /mnt/a2/d3 (4.46TB avail) - /mnt/a3/d2 (1.69TB avail) - /mnt/a3/d3 (1.69TB avail) - /mnt/a3/d4 (1.69TB avail) - /mnt/a3/d1 (1.69TB avail) 
Aug 08 04:23:17 DEBUG write:   Saving 2 metadata files for TV/1234
Aug 08 04:23:17 DEBUG write:     Saving metadata in /mnt/a1/d1/.gh_metastore/TV/1234
Aug 08 04:23:17 DEBUG write:     Saving metadata in /mnt/a1/d3/.gh_metastore/TV/1234
Aug 08 04:23:17 DEBUG write:   Copying 0.00B file to /mnt/a1/d3/TV/1234
Aug 08 04:23:17 DEBUG write:     Creating symlink in share pointing to the above file copy.
Aug 08 04:23:17 DEBUG write:   Saving 2 metadata files for TV/1234
Aug 08 04:23:17 DEBUG write:     Saving metadata in /mnt/a1/d1/.gh_metastore/TV/1234
Aug 08 04:23:17 DEBUG write:     Saving metadata in /mnt/a1/d3/.gh_metastore/TV/1234
Aug 08 04:23:17 DEBUG write:   Copying 0.00B file to /mnt/a1/d1/TV/1234
Aug 08 04:23:17 DEBUG write:   Saving 2 metadata files for TV/1234
Aug 08 04:23:17 DEBUG write:     Saving metadata in /mnt/a1/d1/.gh_metastore/TV/1234
Aug 08 04:23:17 DEBUG write:     Saving metadata in /mnt/a1/d3/.gh_metastore/TV/1234
Aug 08 04:23:17 DEBUG read_smb_spool: Processing Samba spool...
Aug 08 04:23:17 DEBUG sleep: Nothing to do... Sleeping.
Aug 08 04:23:27 DEBUG sleep: Nothing to do... Sleeping.

```

I don't know if this is a greyhole issue or an issue with how OSx deals with smb shares.

**Comment by gboudreau** [comment]
Using Terminal, it logs just one write operation.

My guess is this is just how Finder works with remote shares. (When I tried, I just saw two write operations, no delete op between them.) My guess would be that Finder first tries to create an empty placeholder file, to confirm that writing will work, before it starts sending the data to the remote computer to create the file a second time.

**Comment by Burden50** [comment]
I tried one with a larger file just for fun it looks like it writing the file twice but it could just be logging it wrong. It still seems very fast so I don't think its actually writing it twice.

```
Aug 09 00:35:55 DEBUG read_smb_spool: Processing Samba spool...
Aug 09 00:35:55 DEBUG read_smb_spool: Found 3 new tasks in spool.


Aug 09 00:35:55 INFO write: Now working on task ID 906: write TV/Nintendo - Super Nintendo Entertainment System.rar
Aug 09 00:35:55 INFO write: File created: TV/Nintendo - Super Nintendo Entertainment System.rar - 280MB
Aug 09 00:35:55 DEBUG write: Loading metafiles for TV/Nintendo - Super Nintendo Entertainment System.rar ...
Aug 09 00:35:55 DEBUG write:   Got 0 metadata files.
Aug 09 00:35:55 DEBUG write: 0 metafiles loaded.
Aug 09 00:35:55 DEBUG write:   Drives with available space: /mnt/a1/d4 (4.46TB avail) - /mnt/a2/d4 (4.46TB avail) - /mnt/a1/d2 (4.46TB avail) - /mnt/a1/d3 (4.46TB avail) - /mnt/a2/d1 (4.46TB avail) - /mnt/a1/d1 (4.46TB avail) - /mnt/a2/d3 (4.46TB avail) - /mnt/a2/d2 (4.46TB avail) - /mnt/a3/d2 (1.69TB avail) - /mnt/a3/d3 (1.69TB avail) - /mnt/a3/d4 (1.69TB avail) - /mnt/a3/d1 (1.69TB avail) 
Aug 09 00:35:55 DEBUG write:   Saving 2 metadata files for TV/Nintendo - Super Nintendo Entertainment System.rar
Aug 09 00:35:55 DEBUG write:     Saving metadata in /mnt/a1/d4/.gh_metastore/TV/Nintendo - Super Nintendo Entertainment System.rar
Aug 09 00:35:55 DEBUG write:     Saving metadata in /mnt/a2/d4/.gh_metastore/TV/Nintendo - Super Nintendo Entertainment System.rar
Aug 09 00:35:55 DEBUG write:   Copying 291MB file to /mnt/a2/d4/TV/Nintendo - Super Nintendo Entertainment System.rar
Aug 09 00:35:55 WARN write:     Failed file copy. Will mark this metadata file 'Gone'.
Aug 09 00:35:55 DEBUG write:   Saving 2 metadata files for TV/Nintendo - Super Nintendo Entertainment System.rar
Aug 09 00:35:55 DEBUG write:     Saving metadata in /mnt/a1/d4/.gh_metastore/TV/Nintendo - Super Nintendo Entertainment System.rar
Aug 09 00:35:55 DEBUG write:     Saving metadata in /mnt/a2/d4/.gh_metastore/TV/Nintendo - Super Nintendo Entertainment System.rar
Aug 09 00:35:55 WARN write:     Failed file copy (cont). Will try to re-process this write task, since the source file seems intact.
Aug 09 00:35:55 DEBUG read_smb_spool: Processing Samba spool...


Aug 09 00:35:55 INFO write: Now working on task ID 0: write TV/Nintendo - Super Nintendo Entertainment System.rar
Aug 09 00:35:55 INFO write: File created: TV/Nintendo - Super Nintendo Entertainment System.rar - 312MB
Aug 09 00:35:55 DEBUG write: Loading metafiles for TV/Nintendo - Super Nintendo Entertainment System.rar ...
Aug 09 00:35:55 DEBUG write:   Got 0 metadata files.
Aug 09 00:35:55 DEBUG write: 0 metafiles loaded.
Aug 09 00:35:55 DEBUG write:   Drives with available space: /mnt/a1/d4 (4.46TB avail) - /mnt/a2/d4 (4.46TB avail) - /mnt/a1/d2 (4.46TB avail) - /mnt/a1/d3 (4.46TB avail) - /mnt/a2/d1 (4.46TB avail) - /mnt/a1/d1 (4.46TB avail) - /mnt/a2/d3 (4.46TB avail) - /mnt/a2/d2 (4.46TB avail) - /mnt/a3/d2 (1.69TB avail) - /mnt/a3/d3 (1.69TB avail) - /mnt/a3/d4 (1.69TB avail) - /mnt/a3/d1 (1.69TB avail) 
Aug 09 00:35:55 DEBUG write:   Saving 2 metadata files for TV/Nintendo - Super Nintendo Entertainment System.rar
Aug 09 00:35:55 DEBUG write:     Saving metadata in /mnt/a1/d4/.gh_metastore/TV/Nintendo - Super Nintendo Entertainment System.rar
Aug 09 00:35:55 DEBUG write:     Saving metadata in /mnt/a2/d4/.gh_metastore/TV/Nintendo - Super Nintendo Entertainment System.rar
Aug 09 00:35:55 DEBUG write:   Copying 317MB file to /mnt/a2/d4/TV/Nintendo - Super Nintendo Entertainment System.rar
Aug 09 00:35:56 WARN write:     Failed file copy. Will mark this metadata file 'Gone'.
Aug 09 00:35:56 DEBUG write:   Saving 2 metadata files for TV/Nintendo - Super Nintendo Entertainment System.rar
Aug 09 00:35:56 DEBUG write:     Saving metadata in /mnt/a1/d4/.gh_metastore/TV/Nintendo - Super Nintendo Entertainment System.rar
Aug 09 00:35:56 DEBUG write:     Saving metadata in /mnt/a2/d4/.gh_metastore/TV/Nintendo - Super Nintendo Entertainment System.rar
Aug 09 00:35:56 ERROR write:     Failed file copy (cont). We already retried this task. Aborting.
Aug 09 00:35:56 DEBUG read_smb_spool: Processing Samba spool...


Aug 09 00:35:56 INFO unlink: Now working on task ID 907: unlink TV/Nintendo - Super Nintendo Entertainment System.rar
Aug 09 00:35:56 INFO unlink: File deleted: /mnt/lz/TV/Nintendo - Super Nintendo Entertainment System.rar
Aug 09 00:35:56 DEBUG unlink:   File still exists in landing zone; a new file replaced the one deleted here. Skipping.
Aug 09 00:35:56 DEBUG read_smb_spool: Processing Samba spool...


Aug 09 00:35:56 INFO write: Now working on task ID 908: write TV/Nintendo - Super Nintendo Entertainment System.rar
Aug 09 00:35:56 DEBUG write:   File is still being written to (339MB). Postponing.
Aug 09 00:35:56 DEBUG read_smb_spool: Processing Samba spool...
Aug 09 00:35:56 DEBUG sleep: Nothing to do... Sleeping.
Aug 09 00:36:06 DEBUG read_smb_spool: Processing Samba spool...


Aug 09 00:36:06 INFO write: Now working on task ID 908: write TV/Nintendo - Super Nintendo Entertainment System.rar
Aug 09 00:36:06 DEBUG write:   File is still being written to (1.19GB). Postponing.
Aug 09 00:36:06 DEBUG read_smb_spool: Processing Samba spool...
Aug 09 00:36:06 DEBUG sleep: Nothing to do... Sleeping.


Aug 09 00:36:16 DEBUG read_smb_spool: Processing Samba spool...
Aug 09 00:36:16 INFO write: Now working on task ID 908: write TV/Nintendo - Super Nintendo Entertainment System.rar
Aug 09 00:36:16 DEBUG write:   File is still being written to (2.12GB). Postponing.
Aug 09 00:36:16 DEBUG read_smb_spool: Processing Samba spool...
Aug 09 00:36:16 DEBUG sleep: Nothing to do... Sleeping.
Aug 09 00:36:26 DEBUG read_smb_spool: Processing Samba spool...


Aug 09 00:36:26 INFO write: Now working on task ID 908: write TV/Nintendo - Super Nintendo Entertainment System.rar
Aug 09 00:36:26 INFO write: File created: TV/Nintendo - Super Nintendo Entertainment System.rar - 2.78GB
Aug 09 00:36:26 DEBUG write: Loading metafiles for TV/Nintendo - Super Nintendo Entertainment System.rar ...
Aug 09 00:36:26 DEBUG write:   Got 0 metadata files.
Aug 09 00:36:26 DEBUG write: 0 metafiles loaded.
Aug 09 00:36:26 DEBUG write:   Drives with available space: /mnt/a1/d4 (4.46TB avail) - /mnt/a2/d4 (4.46TB avail) - /mnt/a1/d2 (4.46TB avail) - /mnt/a1/d3 (4.46TB avail) - /mnt/a2/d1 (4.46TB avail) - /mnt/a2/d3 (4.46TB avail) - /mnt/a2/d2 (4.46TB avail) - /mnt/a1/d1 (4.46TB avail) - /mnt/a3/d2 (1.69TB avail) - /mnt/a3/d3 (1.69TB avail) - /mnt/a3/d4 (1.69TB avail) - /mnt/a3/d1 (1.69TB avail) 
Aug 09 00:36:26 DEBUG write:   Saving 2 metadata files for TV/Nintendo - Super Nintendo Entertainment System.rar
Aug 09 00:36:26 DEBUG write:     Saving metadata in /mnt/a1/d4/.gh_metastore/TV/Nintendo - Super Nintendo Entertainment System.rar
Aug 09 00:36:26 DEBUG write:     Saving metadata in /mnt/a2/d4/.gh_metastore/TV/Nintendo - Super Nintendo Entertainment System.rar
Aug 09 00:36:26 DEBUG write:   Copying 2.78GB file to /mnt/a2/d4/TV/Nintendo - Super Nintendo Entertainment System.rar
Aug 09 00:36:32 DEBUG write:     Copy created at 474.2 MBps.
Aug 09 00:36:32 DEBUG write:     Creating symlink in share pointing to the above file copy.
Aug 09 00:36:32 DEBUG write:   Saving 2 metadata files for TV/Nintendo - Super Nintendo Entertainment System.rar
Aug 09 00:36:32 DEBUG write:     Saving metadata in /mnt/a1/d4/.gh_metastore/TV/Nintendo - Super Nintendo Entertainment System.rar
Aug 09 00:36:32 DEBUG write:     Saving metadata in /mnt/a2/d4/.gh_metastore/TV/Nintendo - Super Nintendo Entertainment System.rar
Aug 09 00:36:32 DEBUG write:   Copying 2.78GB file to /mnt/a1/d4/TV/Nintendo - Super Nintendo Entertainment System.rar
Aug 09 00:36:38 DEBUG write:     Copy created at 474.2 MBps.
Aug 09 00:36:43 DEBUG write:   Saving 2 metadata files for TV/Nintendo - Super Nintendo Entertainment System.rar
Aug 09 00:36:43 DEBUG write:     Saving metadata in /mnt/a1/d4/.gh_metastore/TV/Nintendo - Super Nintendo Entertainment System.rar
Aug 09 00:36:43 DEBUG write:     Saving metadata in /mnt/a2/d4/.gh_metastore/TV/Nintendo - Super Nintendo Entertainment System.rar
Aug 09 00:36:43 DEBUG read_smb_spool: Processing Samba spool...
Aug 09 00:36:43 DEBUG sleep: Nothing to do... Sleeping.
```
